data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/bulk/AccumulatingBulkRequest.java

```diff
@@ -24,6 +24,8 @@
 
     int getOperationsCount();
 
+    int incrementAndGetRetryCount();
```

Comment:
I think this is not the right place to have this.

The purpose of this class is to accumulate documents to know when to flush. This is a different concern than handling retries.

Also, each new retry may issue a new `AccumulatingBulkRequest` which would lose the count.